### PR TITLE
Tweak header template.

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @thomasmarsh

### CODE REVIEW

File: src/tr/jsonschema/templates_cpp/class.h.mako [hunk 1]
```diff
@@ -77,8 +77,6 @@ class ${class_name + ((' : public ' + superClass) if superClass else '')}
 public:
     ALIAS_PTR_TYPES(${class_name});
 
-    // TODO: We should eventually make the destructor private to ensure we
-    //       only allocate instances of this class using std::make_shared().
```
Comment (thomasmarsh) [inline src/tr/jsonschema/templates_cpp/class.h.mako]: I'd go ahead and delete the destructor too since we can just accept the default implementation.

**Comment by thomasmarsh** [comment]
LGTM.

File: src/tr/jsonschema/templates_cpp/class.h.mako (hunk 1 above)
Comment (charlesmchen) [inline src/tr/jsonschema/templates_cpp/class.h.mako]: I'm sorry, for some reason this comment showed up after your LGTM so I've already merged. I'll remove this.

Comment (charlesmchen) [inline src/tr/jsonschema/templates_cpp/class.h.mako]: @thomasmarsh See #16.

**Pull request merged**